frontend/views/containers/chatroom/MessageBase.vue

```diff
@@ -100,6 +109,58 @@ export default ({
     isSameSender: Boolean,
     isCurrentUser: Boolean
   },
+  computed: {
+    ...mapGetters(['chatRoomUsers', 'ourUsername']),
+    textObjects () {
+      if (!this.text.includes('@')) {
+        return [{ type: textObjectType.Text, text: this.text }]
+      }
+      const ableMentionings = [
+        ...Object.keys(this.chatRoomUsers).map(u => makeMentionFromUsername(u).me),
+        makeMentionFromUsername('').all
+      ]
+
+      let count = (this.text.match(/@/g)).length
+      const splitByMentionings = (objTexts, mentionings) => {
+        if (!mentionings.length || !count) {
+          return objTexts
+        }
+
+        const curMention = mentionings.pop()
+
+        objTexts = objTexts.map(o => {
+          if (o.type === textObjectType.Mention) {
+            return o
+          } else {
+            if (!o.text.includes(curMention)) {
+              return o
+            } else {
+              return o.text.split(curMention).map((text, i) => {
+                if (!i) {
+                  return { type: textObjectType.Text, text }
+                } else {
+                  count--
+                  const myMention = makeMentionFromUsername(this.ourUsername)
+                  return [
+                    {
+                      type: textObjectType.Mention,
+                      text: curMention,
+                      toMe: curMention === myMention.me || curMention === myMention.all
+                    },
+                    { type: textObjectType.Text, text }
+                  ]
+                }
+              }).flat()
+            }
+          }
+        }).flat()
+
+        return splitByMentionings(objTexts, mentionings)
+      }
+
+      return splitByMentionings([{ type: textObjectType.Text, text: this.text }], ableMentionings)
```

Comment:
This code seems really convoluted and complicated. Is there are simpler solution?

One possibly simpler method would be to use a regex to split the text:

```
> text = 'hello @bob, is your email bob@example.com? my friend @chris wants to chat with you @bob.'
'hello @bob, is your email bob@example.com? my friend @chris wants to chat with you @bob.'
> text.split(/((?:^|\s)@[^\s]+\b)/)
[
  'hello',
  ' @bob',
  ', is your email bob@example.com? my friend',
  ' @chris',
  ' wants to chat with you',
  ' @bob',
  '.'
]
```

Then you can check if the array elements are proper @mentions.

Reply:
@Silver-IT Event better Regex example:

```
> allMentions = ['@bob', '@chris', '@all']
[ '@bob', '@chris', '@all' ]
> text = '@bob hello @bob, is your email bob@example.com? my friend @chris wants to chat with you @bob. @bob? are you there? what do you think of @bob@foo.com?'
'@bob hello @bob, is your email bob@example.com? my friend @chris wants to chat with you @bob. @bob? are you there? what do you think of @bob@foo.com?'
> text.split(new RegExp(`(${allMentions.join('|')})`))
[
  '',
  '@bob',
  ' hello ',
  '@bob',
  ', is your email bob@example.com? my friend ',
  '@chris',
  ' wants to chat with you ',
  '@bob',
  '. ',
  '@bob',
  '? are you there? what do you think of ',
  '@bob',
  '@foo.com?'
]
```